gulpfile.js

```diff
@@ -0,0 +1,214 @@
+const gulp = require('gulp');
+const replace = require('gulp-replace');
+const rename = require('gulp-rename');
+const logger = require('gulplog');
+const path = require('path');
+const fs = require('mz/fs');
+const del = require('del');
+const mergeStream = require('merge-stream');
+const {forkStream, PolymerProject, addServiceWorker, HtmlSplitter, getOptimizeStreams} = require('polymer-build');
+const projectConfig = require('./polymer.json');
+
+
+/* ********** HELPERS ****************************************************** */
+
+/**
+ * Returns a promise that resolves when the stream ends,
+ * or rejects when the stream throws an error
+ */
+const waitFor = (stream) => new Promise((resolve, reject) => {
+  stream.on('end', resolve);
+  stream.on('error', reject);
+});
+
+/**
+ * Utility that allows to avoid loads of .pipe chains
+ */
+const pipeStreams = (streams) => Array.prototype.concat.apply([], streams)
+  .reduce((a, b) => a.pipe(b));
+
+/**
+ * This helper function does basically the same things as the Polymer CLI
+ * `build` command, but it also has some extra options to help us with some
+ * special tasks (e.g. renaming the node_modules output folder).
+ */
+const build = async (options, polymerProject) => {
+  const buildName = options.name || 'default';
+  const buildDirectory = path.join(options.outputDir, buildName);
+  const logPrefix = options.logPrefix || '';
+
+  logger.info(`${logPrefix}(${buildName}) Preparing...`);
```

Comment:
why can't we just `console.log`?

Reply:
Code from polymer-cli. Does nice formatting stuff, I think.

Reply:
i guess my biggest comment is why do we have to redo all the logic `polymer build` does instead of...exec-ing it.